### PR TITLE
Add more error logs for sync controller & kubefedcluster controller

### LINKED ISSUE #1407: Display more errorr logs when reconciling fed resources fails
**What happened**:
https://github.com/kubernetes-sigs/kubefed/blob/5ee53350e44082e8f586dd10b2c17de1599398c5/pkg/controller/sync/controller.go#L283
When failed to add  a finalizer to a fedresource, `RecordError ` function is called. However, it only creates an event and there is nothing showing in `kubefed controller manager` logs, which can be difficult to debug why reconciling failed.
**What you expected to happen**:
The logs output more error messages, like what others do.
https://github.com/kubernetes-sigs/kubefed/blob/5ee53350e44082e8f586dd10b2c17de1599398c5/pkg/controller/sync/controller.go#L249
**How to reproduce it (as minimally and precisely as possible)**:

**Anything else we need to know?**:

**Environment**:
- Kubernetes version (use `kubectl version`) v1.19
- KubeFed v0.7.0
- Scope of installation (namespaced or cluster) cluster
- Others

/kind bug

### PR DESCRIPTION
**Pull request by yuswift**
**What this PR does / why we need it**:
Add more error logs for `sync controller` & `kubefedcluster controller`, which can be useful to debug.
**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #1407

### CODE REVIEW

**Comment by yuswift** [comment]
@irfanurrehman could you take a look at this pr when you are free?

**Comment by irfanurrehman** [comment]
Thanks for doing this @yuswift 
/lgtm

File: pkg/controller/kubefedcluster/controller.go [hunk 1]
```diff
@@ -184,7 +184,9 @@ func (cc *ClusterController) addToClusterSet(obj *fedv1b1.KubeFedCluster) {
 	// create the restclient of cluster
 	restClient, err := NewClusterClientSet(obj, cc.client, cc.fedNamespace, cc.clusterHealthCheckConfig.Timeout)
 	if err != nil || restClient == nil {
-		cc.RecordError(obj, "MalformedClusterConfig", errors.Wrap(err, "The configuration for this cluster may be malformed"))
+		cc.RecordError(obj, "MalformedClusterConfig", errors.Wrap(err, "The configuration for this "+
```
Comment (hectorj2f) [inline pkg/controller/kubefedcluster/controller.go:187]: Why are you concating this string here ?

File: pkg/controller/kubefedcluster/controller.go [hunk 2]
```diff
@@ -184,7 +184,9 @@ func (cc *ClusterController) addToClusterSet(obj *fedv1b1.KubeFedCluster) {
...
+			"cluster may be malformed"))
+		utilruntime.HandleError(errors.Wrapf(err, "the configuration for this cluster may be malformed"))
```
Comment (hectorj2f) [inline pkg/controller/kubefedcluster/controller.go:189]: Why don't you use klog ?

Comment (hectorj2f) [reply pkg/controller/kubefedcluster/controller.go:189]: We are not using utilruntime anywhere else in the code.

Comment (yuswift) [reply pkg/controller/kubefedcluster/controller.go:189]: here we used it
https://github.com/kubernetes-sigs/kubefed/blob/4859cebef7f11e61277a76b61e1f6a8189fcee63/pkg/controller/kubefedcluster/controller.go#L195
if you think using utilruntime is not necessary, I will change it to klog.

**Review by hectorj2f — COMMENTED** [review]
/lgtm

**Pull request merged**